### DISCUSSION TITLE
Is it possible to use a property inside a script?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#861] [Q&A] [answered]
Hi

Is it possible to pass the value of a property to a script (javascript) added on html page?

Let's say I have a
`<div property="message"></div>`

And I want to use that 'message' property value inside the following code:

```
<script type="text/javascript">
    alert("[message]") ;
</script>
```
(this code doesn't work) 

Codepen :
https://codepen.io/galinhalx/pen/zYRjxGp

**Comment by LeaVerou** [comment, marked as answer]
Using expressions in `<script>` tags wouldn't work, because by the time Mavo resolves the expressions, the browser has already tried to execute the script.

However, you can define global functions in JS, and then call them in Mavo expressions, which is typically the easiest way to pass property data to JS. HTML attribute event handlers (e.g. `onclick`) also work.

**Comment by GalinhaLX** [comment]
Hi @LeaVerou 

I've been able to use it with onclick, 

but not defining a JS function inside my own JS file, or even inside my HTML page

Do they need to be written inside mavo.js file?

Is there a demo somewhere about defining a global function that can be used by Mavo?

**Reply by DmitrySharabin** [reply]
I think this one should help: https://codepen.io/leaverou/pen/oNzvaaP

**Comment by GalinhaLX** [comment]
Wow ! , I even don't understand where I was wrong.... 

Thank you so much Dmitry
